components/gitpod-db/go/cost_center.go

```diff
@@ -209,17 +209,23 @@ func (c *CostCenterManager) IncrementBillingCycle(ctx context.Context, attributi
 		log.Infof("Cost center %s is not yet expired. Skipping increment.", attributionId)
 		return cc, nil
 	}
-	billingCycleStart := NewVarCharTime(now)
+	billingCycleStart := now
 	if cc.NextBillingTime.IsSet() {
-		billingCycleStart = cc.NextBillingTime
+		billingCycleStart = cc.NextBillingTime.Time()
+	}
+	nextBillingTime := billingCycleStart.AddDate(0, 1, 0)
+	for nextBillingTime.Before(now) {
```

Comment:
Should this be an `if` so it doesn't loop infinitely?

Reply:
Ah nevermind, you set `nextBillingTime` below - it's early still for me 😄

Reply:
it iterates until `nextBillingTime` is in the future